app/controllers/reviews_controller.rb

```diff
@@ -20,10 +20,16 @@ def edit
   end
 
   def update
-    review = Review.find(params[:id])
-    review.update(review_params)
-    review.save
-    redirect_to "/shelters/#{params[:shelter_id]}"
+    # @shelter = Shelter.find(params[:shelter_id])
+    @review = Review.find(params[:id])
+    @review.update(review_params)
+    if @review.save
+      redirect_to "/shelters/#{params[:shelter_id]}"
+    else
+      flash[:notice] = "Review not updated: Reviews must have a title, rating, and content"
+      render :edit
+      # redirect_to "/shelters/#{@shelter.id}/reviews/#{review.id}/edit"
```

Comment:
This seems to work. All the tests are passing but I have not tested manually.

redirect => routes => reviews#edit

I am not 100% sure but I believe render edit goes directly to displaying the edit view without going through routes and the action.

Reply:
One thing to note @review NEEDs to be an instance variable. If it is review then the edit view will be rendered but since it requires @review it will break if the variable isn't an instance variable.

Reply:
After our class today I think we should just stick with the redirect.